Issues #38 and #39 fixes - Work in progress

latin-programmer opened the pull request:
Hi pongstylin, 

I've been working on this off and on and I got stuck after I stopped getting errors. Classic up seems to be back in working order. I am no longer getting console errors for face-off and chaos apps, but those two are still wonky. 

Overall, my goal was to per our discussion to bring the "data" property concept to the local transport and have the apps match it. This took care of the errors but I am still having some issues getting the bots to talk to the human player properly (and the worker sometimes just stops receiving messages). 

Of course, I want everything to be back in 100% working order but I decided it would be best to show you what I have and get some guidance on what I'm missing. Heck, you might even see one or two lines of code that would get them back in order. -El Cid

pongstylin commented on src/chaos-app.js:
```diff
@@ -10,8 +10,9 @@ Tactics.App = (function ($, window, document) {
     {assignment:[5, 1], type:'Knight'},
     {assignment:[6, 1], type:'Knight'},
   ];
-  var gameData = {
```
@latin-programmer Nesting the data inside of a "data" tag shouldn't be necessary and should be reverted.  The so-called "gameData" variable would probably be better named as "gameStateData" since it will ultimately be consumed by the "tactics/GameState.js" class when loading an instance.

pongstylin commented on src/chaos-app.js:
```diff
@@ -10,8 +10,9 @@ Tactics.App = (function ($, window, document) {
   ];
-  var gameData = {
-    type: 'Chaos',
+  var gameData = { data: {} };
```
@latin-programmer Nesting the data inside of a "data" tag shouldn't be necessary and should be reverted. The so-called "gameData" variable would probably be better named as "gameStateData" since it will ultimately be consumed by the "tactics/GameState.js" class when loading an instance.

pongstylin commented on src/classic-app.js:
```diff
@@ -20,7 +20,8 @@ Tactics.App = (function ($, window, document) {
     {assignment:[9, 2], type:'Scout'},
   ];
-  var gameData = {
+  var gameData = { data: {} };
```
@latin-programmer Same here.  Nesting is bad.

pongstylin commented on src/faceoff-app.js:
```diff
@@ -10,7 +10,8 @@ Tactics.App = (function ($, window, document) {
     {assignment:[6, 2], type:'Knight'},
   ];
-  var gameData = {
+  var gameData = { data: {} };
```
@latin-programmer Same here, nesting is bad.

pongstylin commented on src/tactics/Bot.js:
```diff
@@ -12,7 +12,7 @@ export default class {
      */
     let startTurnListener = event => {
-      if (event.teamId === this.team.id)
+      if (event.data.teamId === this.team.id)
```
Good!

pongstylin commented on src/tactics/LocalTransport.js:
```diff
@@ -27,14 +27,14 @@ export default class LocalTransport {
   static createGame(gameData) {
     let transport = new LocalTransport();
-    transport._post({ type:'create', data:gameData });
+    transport._post({ type:'create', data:gameData.data });
```
@latin-programmer Once you stop nesting the data, you can revert this change as well.

pongstylin commented on src/tactics/LocalTransport.js:
```diff
@@ -27,14 +27,14 @@ export default class LocalTransport {
   static loadGame(gameData) {
     let transport = new LocalTransport();
-    transport._post({ type:'load', data:gameData });
+    transport._post({ type:'load', data:gameData.data });
```
@latin-programmer This can be reverted as well.

pongstylin commented on src/tactics/LocalTransport.js:
```diff
@@ -148,28 +148,29 @@ export default class LocalTransport {
-          Object.assign(this._data, {
-            currentTurnId: turnId,
-            currentTeamId: teamId,
+        .on('startTurn', ({ data }) => {
```
Good!

pongstylin commented on src/tactics/LocalTransport.js:
```diff
@@ -148,28 +148,29 @@ export default class LocalTransport {
-            currentTurnId: turnId,
-            currentTeamId: teamId,
-            actions:       actions,
+        .on('reset', ({data}) => {
```
@latin-programmer The 'reset' event no longer exists.  I have renamed it to the "revert" event.

pongstylin commented on src/tactics/core.js:
```diff
@@ -65,7 +65,7 @@ window.Tactics = (function () {
     createLocalGame: function (stateData) {
       let transport = LocalTransport.createGame(stateData);
-      let localTeamIds = stateData.teams
+      let localTeamIds = stateData.data.teams
```
@latin-programmer This can be reverted once data is no longer nested.

pongstylin commented:
@latin-programmer I finished reviewing the code changes.  So, when you're ready, please resolve the comments I had.

latin-programmer commented:
@pongstylin Thank you for the review. I addressed all the issues in the code review changes. The apps are still in the same state with the new changes (so more work will need to be done), but overall I'm still good with sending what I have so far as it fixes the classic app and removes the errors I reported for the other two apps. The problems remaining are: 
 - Face off app loads but does not work (the bot may start a turn or may not, but then you cannot play after) 
 - Chaos app is missing the egg and you cannot play after.
